### DISCUSSION TITLE
SxB Defeat expansion

### DISCUSSION THREAD

**Discussion by XspeedPL** [ErinaSugino/Starbound-SBR-Mod#47] [Ideas]
After defeating enemies, being able to interact with them only for a period of time before they despawn feels like a waste.

Instead there could be a slavery mechanism implemented, with collar and chain item (think capture pods, but keeps NPC/monster close to player) and tether station object (same as the one for pets).

While chained to player they could be used as meat shields in combat (dealing knockback?), and while tethered forcefully turned to nodes (by interacting with the tether station).

Sub-idea: possibility to turn those NPCs into followers after using them for a while.

**Comment by ErinaSugino** [comment]
That sounds like an awful lot of work. LEt's first sedate the elephant in the room and get defeat as-is working. That is, one day. In the future.

This also sounds vaguely like a feature from the Dom-Sub mod that one ambitious guy on Discord is apparently trying to do, so might look forward to that first.

**Reply by DraconicChaos901** [reply]
So I actually tested defeat myself very recently and out the gate, it actually DOES seem to mostly work with Reborn. The only things that don't work as originally intended is that the UI still appears when defeat has code telling it not to (which I'm currently looking to find a fix myself). Another issue I've found is that even if the UI were hidden from the player, there's also the issue of filtering positions and having one randomly chosen when the node is created.

**Reply by ErinaSugino** [reply]
For every person that defeat works for, there is atleast one person that it doesn't work for. It's the nature of all the moving parts in it. The damage handler doesn'T properly capture all sources of damage, It triggering on every entity guaranteed turning simple monster killing into a waiting game is an error by design, the placement of nodes especially in dungeons, the back-and-forth communication with surrounding entities to control them into joining, and of course the invincibility glitch. For Defeat to "work properly" and be fixed, all of these need to be solved in a stable manner, so that every user gets the same experience.

**Reply by DraconicChaos901** [reply]
That is very true, but the fact that it's still at least somewhat usable is a miracle enough. I've actually been considering making my own fork of defeat specifically to work with Sexbound Reborn while also fixing some of it's major issues.

**Reply by ErinaSugino** [reply]
Well, good luck with that. A lot of people will lick their fingers waiting on that one.
